### PR TITLE
Added Carting and Catalog Projects

### PR DESCRIPTION
**Pull request by Jesus-Alanis**

### CODE REVIEW

File: Carting/Domain/Entities/Item.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using Carting.Domain.ValueObjects;
+
+namespace Carting.Domain.Entities
+{
+    public class Item
+    {
+        public long Id { get; set; }
+        public long CartId { get; set; }
+        public string Name { get; set; }
+        public Image Image { get; set; }
```
Comment (dmitry-slabko) [inline Carting/Domain/Entities/Item.cs:10]: This property may also be set as nullable: `Image?`

File: Catalog.Application/DTOs/Category.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Catalog.Application.DTOs
+{
+    public class Category
+    {
+        public string Name { get; set; }
+        public string Url { get; set; }
```
Comment (dmitry-slabko) [inline Catalog.Application/DTOs/Category.cs:6]: For consistency with other DTOs, this property may be declared as nullable: `string?`

File: Catalog.DataAccess/Repositories/CategoryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected void Dispose(bool disposing)
```
Comment (dmitry-slabko) [inline Catalog.DataAccess/Repositories/CategoryRepository.cs:58]: Same comment on simplifying the implementation as for CartRepository.

File: Catalog.DataAccess/Repositories/ItemRepository.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            Dispose(false);
+        }
+        
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+        
+        protected void Dispose(bool disposing)
```
Comment (dmitry-slabko) [inline Catalog.DataAccess/Repositories/ItemRepository.cs:58]: Same comment for simplifying Dispose implementation.

**Review by dmitry-slabko — COMMENTED** [review]
A few general comments:
- please pay attention to reference variable/fields when they may get nulls, and declare them as nullable types accordingly;
- for this level of complexity splitting to separate Repository/Service pair is not really required, though it is fine considering that more logic will be added later.

File: Carting/DataAccess/Repositories/CartRepository.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
```
[the remark this answers is not in the document]
Comment (Jesus-Alanis) [reply Carting/DataAccess/Repositories/CartRepository.cs:56]: done

File: Carting/DataAccess/Repositories/CartRepository.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
+﻿using Carting.Domain.Entities;
+using Carting.Domain.Repositories;
+using LiteDB;
+
+namespace Carting.DataAccess.Repositories
+{
+    internal class CartRepository : ICartRepository
+    {
+        private bool _disposedValue;
+        private ILiteDatabase _database;
```
[the remark this answers is not in the document]
Comment (Jesus-Alanis) [reply Carting/DataAccess/Repositories/CartRepository.cs:10]: done

File: Carting/Domain/Entities/Item.cs (hunk 1 above)
Comment (Jesus-Alanis) [reply Carting/Domain/Entities/Item.cs:10]: done

File: Catalog.Application/DTOs/Category.cs (hunk 2 above)
Comment (Jesus-Alanis) [reply Catalog.Application/DTOs/Category.cs:6]: done

File: Catalog.DataAccess/Repositories/CategoryRepository.cs (hunk 3 above)
Comment (Jesus-Alanis) [reply Catalog.DataAccess/Repositories/CategoryRepository.cs:58]: done

File: Catalog.DataAccess/Repositories/ItemRepository.cs (hunk 4 above)
Comment (Jesus-Alanis) [reply Catalog.DataAccess/Repositories/ItemRepository.cs:58]: done

**Pull request merged**